File: capybara-rspec/pages/flight_page.rb
require File.join(File.dirname(__FILE__), "abstract_page.rb")

class FlightPage < AbstractPage

  def initialize()
    super("") # <= TEXT UNIQUE TO THIS PAGE
  end

  def select_trip_type(trip_type)
    choose("tripType", option: trip_type)
  end

  def select_depart_from(from_port)
    select(from_port, :from => 'fromPort')
  end

  def select_arrive_at(to_port)
    select(to_port, :from => 'toPort')
  end

  def select_departure_day(depart_day)
    select(depart_day, :from => 'departDay')
  end

  def select_departure_month(depart_month)
    select(depart_month, :from => 'departMonth')
  end

  def select_return_date(day, month)
    select(day, :from => 'returnDay')
    select(month, :from => 'returnMonth')
  end

  def click_continue
    click_button("Continue")
  end

end

File: selenium-webdriver-nodejs-jasmine/pages/flight_page.js
const AbstractPage = require('./abstract_page');

var webdriver = require('selenium-webdriver'),
    By = webdriver.By,
    until = webdriver.until;

class FlightPage extends AbstractPage {

  constructor(driver) {
    super(driver);
  }

  async selectTripType(trip_type) {
    await this.driver.findElement(By.xpath("//input[@name='tripType' and @value='" + trip_type + "']")).click();
  }

  async selectDepartFrom(city) {
    await this.driver.findElement(By.name("fromPort")).sendKeys(city);
  }

  async selectArriveAt(city) {
    await this.driver.findElement(By.name("toPort")).sendKeys(city);
  }

  async selectDepartDay(day) {
    await this.driver.findElement(By.name("departDay")).sendKeys(day);
  }

  async selectDepartMonth(month_year) {
    await this.driver.findElement(By.name("departMonth")).sendKeys(month_year);
  }

  async selectReturnDay(day) {
    await this.driver.findElement(By.name("returnDay")).sendKeys(day);
  }

  async selectReturnMonth(month_year) {
    await this.driver.findElement(By.name("returnMonth")).sendKeys(month_year);
  }

  async clickContinue() {
    await this.driver.findElement(By.xpath("//input[@value='Continue']")).click();
  }
};

module.exports = FlightPage;
